networkx/linalg/laplacianmatrix.py

```diff
@@ -147,36 +147,131 @@ def normalized_laplacian_matrix(G, nodelist=None, weight="weight"):
 
 
 @nx._dispatch(edge_attrs="weight")
-def total_spanning_tree_weight(G, weight=None):
+def total_spanning_tree_weight(G, weight=None, root=None):
     """
     Returns the total weight of all spanning trees of `G`.
 
-    Kirchoff's Tree Matrix Theorem states that the determinant of any cofactor of the
+    Kirchoff's Tree Matrix Theorem [1]_, [2]_ states that the determinant of any cofactor of the
     Laplacian matrix of a graph is the number of spanning trees in the graph. For a
     weighted Laplacian matrix, it is the sum across all spanning trees of the
     multiplicative weight of each tree. That is, the weight of each tree is the
     product of its edge weights.
 
+    For unweighted graphs, the total weight equals the number of spanning trees in `G`.
+
+    For directed graphs, the total weight follows by summing over all directed spanning trees in `G`
+    that end in the `root` node [3]_.
+
```

Comment:
Change end -> start but also add a warning of the deprecation to the doc_string with same message as in the deprecation warning.
```suggestion
    that start in the `root` node [3]_.

    .. deprecated:: 3.5

            `total_spanning_tree_weight` is deprecated and will be removed in v3.5.
            Use `nx.number_of_spanning_trees(G)` instead.

```